File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    
    if(typeof list1 === 'undefined' && typeof list2 === 'undefined'){
        return
    }else if(list1 === null){
        return list2
    }else if(list2 === null){
        return list1
    };
    
    let out = new ListNode();
    let node= out;
    
    while(list1 !== null && list2 !== null){
        
        if(list1.val <= list2.val){
            node.val = list1.val;
            list1 = list1.next;
        }else{
            node.val = list2.val;
            list2 = list2.next
        };
        
        if(list1 !== null && list2 !== null){
            node.next = new ListNode();
            node = node.next;
        }
        
        
    }
    
    if(list1 !== null){
        node.next = list1;
    }else if(list2 !== null){
        node.next = list2;
    }
    return out
    
};

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        
        if list1 is None and list2 is None:
            return None
        elif list1 is None:
            return list2
        elif list2 is None:
            return list1
        
        out = ListNode()
        node = out
        
        while list1 is not None and list2 is not None:
            
            if list1.val <= list2.val:
                node.val = list1.val
                # node.next = ListNode()
                list1 = list1.next
                
            else:
                node.val = list2.val
                # node.next = ListNode()
                
                list2 = list2.next 
             
            
            if list1 is not None and list2 is not None:
                node.next = ListNode()
                node = node.next
        
        if list1 is not None:
            node.next = list1
        if list2 is not None:
            node.next = list2
            
        return out
    
        